### DISCUSSION TITLE
What impacts the build time?

### DISCUSSION THREAD

**Discussion by JeremieLitzler** [CaiJimmy/hugo-theme-stack#975] [General]
Hi @CaiJimmy,

I have a question for you: today, I noticed that my build time just reached 30s on one of my blogs. While, I've found that I just needed to set the timeout config to 60s for now, I'd love to know your opinion about what has the most impact on the build time?

Also, on my PC (i3 and 8GB), the `hugo` command runs in less than 10s.
For the exact same repo and commit, it runs at 36s on Netlify.

Do you have any other advice?

Here is the error I got from Netlify this morning:

```
8:12:14 PM: hugo v0.123.8-5fed9c591b694f314e5939548e11cc3dcb79a79c+extended linux/amd64 BuildDate=2024-03-07T13:14:42Z VendorInfo=gohugoio
8:12:48 PM: ERROR render of "page" failed: "/opt/build/repo/themes/hugo-theme-stack/layouts/_default/single.html:27:7": execute of template failed: template: _default/single.html:27:7: executing "main" at <partial "article/article.html" .>: error calling partial: partial "article/article.html" timed out after 30s. This is most likely due to infinite recursion. If this is just a slow template, you can try to increase the "timeout" config setting.
8:12:48 PM: Total in 33961 ms
8:12:48 PM: Error: error building site: render: failed to render pages: render of "page" failed: "/opt/build/repo/themes/hugo-theme-stack/layouts/_default/single.html:27:7": execute of template failed: template: _default/single.html:27:7: executing "main" at <partial "article/article.html" .>: error calling partial: partial "article/article.html" timed out after 30s. This is most likely due to infinite recursion. If this is just a slow template, you can try to increase the "timeout" config setting.
```

Here the stats of my blog repo:

```
8:25:17 PM:                    |  EN
8:25:17 PM: -------------------+-------
8:25:17 PM:   Pages            |  347
8:25:17 PM:   Paginator pages  |   68
8:25:17 PM:   Non-page files   |  499
8:25:17 PM:   Static files     |  113
8:25:17 PM:   Processed images | 1331
8:25:17 PM:   Aliases          |  110
8:25:17 PM:   Cleaned          |    0
8:25:17 PM: Total in 36602 ms
```

Thanks for your help.

**Comment by CaiJimmy** [comment]
> For the exact same repo and commit, it runs at 36s on Netlify.

Try setting up cache for `resources` folder (https://www.netlify.com/integrations/community-built/hugo-cache-resources-build-plugin/). Image processing is the part that takes most of the build time.

**Reply by JeremieLitzler** [reply]
@CaiJimmy 

Does the original image size would impact their processing?

I'll look into the cache of `resources` folder.

Thanks.

**Reply by JeremieLitzler** [reply]
Wooow ! The plugin is wooow, I went from 36s to 750**ms**!

**Reply by CaiJimmy** [reply]
> Does the original image size would impact their processing?

The first build would be slower, but should not be a big problem with the cache.
